### PR TITLE
support puppet 5 on windows

### PR DESCRIPTION
**Pull request by zshahan**
Currently, the module looks at a download url that only contains puppet 3 and puppet 4 packages.  This extra condition helps support puppet 5 packages if puppet_agent::collection is set to puppet5.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -130,6 +130,6 @@
     {"name":"puppetlabs-stdlib","version_requirement":">= 4.6.0 < 5.0.0"},
     {"name":"puppetlabs-transition","version_requirement":">= 0.1.1 < 0.2.0"},
     {"name":"puppetlabs-inifile","version_requirement":">= 1.2.0 < 3.0.0"},
-    {"name":"puppetlabs-apt","version_requirement":">= 2.0.1 <= 4.1.0"}
+    {"name":"puppetlabs-apt","version_requirement":">= 2.0.1"}
```
Comment (Iristyle) [inline metadata.json]: I'm not sure what this has to do with Puppet 5 and Windows?

File: manifests/init.pp [hunk 2]
```diff
@@ -90,8 +90,7 @@
       $_expected_package_version = $package_version
     }
 
-    $aio_upgrade_required = ($is_pe == false and $_expected_package_version != undef) or
-      (getvar('::aio_agent_version') != undef and $_expected_package_version != undef and
+    $aio_upgrade_required = (getvar('::aio_agent_version') != undef and $_expected_package_version != undef and
```
Comment (Iristyle) [inline manifests/init.pp]: A test demonstrating what was fixed would be helpful

**Comment by BarnumD** [comment]
This is still happening with Puppet 6.  Any update?

**Comment by BarnumD** [comment]
I was able to get the puppet agent 6 version installed this way.. So I'm not sure if this pull request is really necessary.
`class {'::puppet_agent':
  package_name    => 'puppet6/puppet-agent',
  package_version => '6.0.4',
}`

**Comment by gavinelder** [comment]
You want to specify the collection 
```
  class {'::puppet_agent':
      package_version => '5.5.17',
      collection      => 'puppet5',
  }
```

**Pull request closed without merging**